tests/tests_e3/os/process/main_test.py

```diff
@@ -228,45 +229,62 @@ def test_interrupt():
 
 @pytest.mark.skipif(psutil is None, reason='require psutil')
 def test_kill_process_tree():
+
+    is_appveyor_test = bool(os.environ.get('APPVEYOR'))
+    wait_timeout = 3
+    if is_appveyor_test:
+        wait_timeout *= 3
     p1 = e3.os.process.Run(
         [sys.executable, '-c',
          'import time; time.sleep(10); import sys; sys.exit(2)'],
         bg=True)
-    e3.os.process.kill_process_tree(p1.pid)
-    p1.wait()
+    e3.os.process.kill_process_tree(p1.pid, timeout=wait_timeout)
     assert p1.status != 2
+    assert not p1.is_running()
 
-    def get_one_child():
-        cmd = [sys.executable, '-c',
-               'import e3.os.process; import time; import sys;'
-               ' e3.os.process.Run([sys.executable, "-c",'
-               ' "import time; time.sleep(10)"]);'
-               'time.sleep(10)']
-
-        p_one_child = e3.os.process.Run(cmd, bg=True)
+    def get_one_child(idx):
+        pid_file = 'child_pid_%d' % idx
+        gen_prog_name = 'child_prog_%d' % idx
+        prog = textwrap.dedent("""\
+            import e3.os.process, os, sys, time
+            child_cmd = "import os, time;"
+            child_cmd += "f = open('%s', 'w');"
+            child_cmd += "f.write(str(os.getpid()));"
+            child_cmd += "f.close();"
+            child_cmd += "time.sleep(60);"
+            e3.os.process.Run([sys.executable, '-c', child_cmd])
+            time.sleep(60)
+            """ % pid_file)
+
+        with open(gen_prog_name, 'w') as f:
+            f.write(prog)
+
+        p_one_child = e3.os.process.Run(
+            [sys.executable, gen_prog_name], bg=True)
         for k in range(0, 100):
-            p_one_child_children = p_one_child.children()
-            if len(p_one_child_children) == 1:
-                break
+            try:
+                with open(pid_file) as f:
+                    child_pid = f.read()
+                    if child_pid:
+                        break
+            except IOError:
+                pass
             time.sleep(0.1)
-        assert len(p_one_child_children) == 1
-        return p_one_child, p_one_child_children
+        e3.fs.rm(gen_prog_name)
```

Comment:
Isn't this file still locked because the process you spawned is still running?

Reply:
No it's not. Or else the check would not have passed on AppVeyor ;)